trie/trieChangesBatch/trieChangesBatch.go

```diff
@@ -0,0 +1,99 @@
+package trieChangesBatch
+
+import (
+	"github.com/multiversx/mx-chain-core-go/core"
+	"sort"
+	"sync"
+)
+
+type trieChangesBatch struct {
+	insertedData map[string]core.TrieData
+	deletedKeys  map[string]struct{}
+
+	mutex sync.RWMutex
+}
+
+// NewTrieChangesBatch creates a new instance of trieChangesBatch
+func NewTrieChangesBatch() *trieChangesBatch {
+	return &trieChangesBatch{
+		insertedData: make(map[string]core.TrieData),
+		deletedKeys:  make(map[string]struct{}),
+	}
+}
+
+// Add adds a new key and data to the batch
+func (t *trieChangesBatch) Add(key []byte, data core.TrieData) {
+	t.mutex.Lock()
+	defer t.mutex.Unlock()
+
+	_, ok := t.deletedKeys[string(key)]
+	if ok {
+		delete(t.deletedKeys, string(key))
+	}
+
+	t.insertedData[string(key)] = data
+}
+
+// Remove removes the key from the batch
+func (t *trieChangesBatch) Remove(key []byte) {
+	t.mutex.Lock()
+	defer t.mutex.Unlock()
+
+	_, ok := t.insertedData[string(key)]
+	if ok {
+		delete(t.insertedData, string(key))
+	}
+
+	t.deletedKeys[string(key)] = struct{}{}
+}
+
+// Get returns the data for the given key
+func (t *trieChangesBatch) Get(key []byte) ([]byte, bool) {
+	t.mutex.RLock()
+	defer t.mutex.RUnlock()
+
+	data, ok := t.insertedData[string(key)]
+	if ok {
+		return data.Value, true
+	}
+
+	_, ok = t.deletedKeys[string(key)]
+	if ok {
+		return nil, true
+	}
+
+	return nil, false
+}
+
+// GetSortedDataForInsertion returns the data sorted for insertion
+func (t *trieChangesBatch) GetSortedDataForInsertion() ([]string, map[string]core.TrieData) {
```

Comment:
I chose to sort this data in preparation for the next PR in which I will add the mutex on the trie node instead of the whole trie struct. In the next PR I will pass the whole batch when updating the trie. If the batch is sorted then we can split the batch each level of the trie we iterate. So for the child at index 0 we give the sub-batch in which all the keys start with 0, for the child at index 1 the sub-batch of keys that start with 1, and so on.